src/app/handlers/command.handler.ts

```diff
@@ -114,42 +144,57 @@ export class CommandHandler extends Handler {
     return { name, args };
   }
 
-  private async _attemptRunPlugin(
-    message: types.IMessage,
-    plugin: types.IPlugin,
-    command: types.ICommand,
-    isDM: boolean
-  ) {
+  private async _attemptRunPlugin(commandData: CommandResolvable) {
+    const { isDM, plugin, message } = commandData;
+
     if ((isDM && !plugin.usableInDM) || (!isDM && !plugin.usableInGuild)) {
       return;
     }
 
     const permissionResponse = plugin.hasPermission(message);
     if (!isDM && permissionResponse !== true) {
-      message.reply(permissionResponse);
+      message.reply({ content: `${permissionResponse}` });
       return;
     }
 
-    if (!plugin.validate(message, command.args)) {
-      await message.reply(`Invalid arguments! Try: \`${Constants.Prefix}${plugin.usage}\``);
-      return;
+    if (commandData.type === 'message') {
+      if (!plugin.validate(commandData.message, commandData.command.args)) {
+        await message.reply(`Invalid arguments! Try: \`${Constants.Prefix}${plugin.usage}\``);
+        return;
+      }
     }
 
     if (!plugin.isActive) {
       await message.reply('This plugin has been deactivated.');
       return;
     }
 
+    const user = message instanceof CommandInteraction ? message.user.tag : message.author.tag;
+
     const pEvent: types.IPluginEvent = {
       status: 'starting',
       pluginName: plugin.name,
-      args: command.args,
-      user: message.author.tag,
+      args: [],
```

Comment:
```suggestion
      args: 
          message instanceof CommandInteraction ? 
              commandData.command.args : commandData.command.options,
```
Would need a good way of `toString`ing the options though, since its basically an unsorted collection

Probably something like this
```ts
commandData.command.options.map((option) => `${option.name}: ${option.value}`)
```
I'm not 100% familiar slash commands yet so not sure if thats fully correct